piff/stats.py

```diff
@@ -20,184 +20,227 @@
-        positions = np.array([ (star.properties['u'], star.properties['v']) for star in stars ])
+        positions = np.array([ (star.data.properties['u'], star.data.properties['v'])
+                               for star in stars ])
 
-        # get target stars with same properies as real stars, but blank image
+        # generate the model stars and measure moments
         if logger:
-            logger.info("Making images of model")
-        stamp_size = np.max( [star.image.array.shape for star in stars ] )
-        test_stars = [ piff.Star(piff.StarData.makeTarget(
-                            properties=star.properties, stamp_size=stamp_size), None)
-                       for star in stars ]
-        test_stars = psf.interp.interpolateList(test_stars)
-
-        # generate the model stars measure moments with Gaussian on
-        # interpolated model image
-        if logger:
-            logger.info("Generating and Measuring Model Stars")
-        shapes_model = np.array([ self.hsm(*psf.model.draw(star).data.getImage())
-                                  for star in test_stars ])
+            logger.debug("Generating and Measuring Model Stars")
+        shapes_model = np.array([ self.hsm(psf.drawStar(star)) for star in stars ])
 
         return positions, shapes_truth, shapes_model
 
-class ShapeStatistics(Statistics):
-    """Returns histograms of shape differences using HSM
-    """
 
-    def __init__(self, psf, stars, logger=None):
+class ShapeHistogramsStats(Stats):
+    """Stats class for calculating histograms of shape residuals
+
+    This will compute the size and shapes of the observed stars and the PSF models and
+    make histograms of both the values and the residuals.
+
+    The plot will have 6 axes.  The top row will have histograms of T, g1, g2, with the model
+    and data color coded.  The bottom row will have histograms of the differences.
+
+    After a call to :func:`compute`, the following attributes are accessible:
+
+        :u:         The u positions in field coordinates.
+        :v:         The v positions in field coordinates.
+        :T:         The size (T = Ixx + Iyy) of the observed stars.
+        :g1:        The g1 component of the shapes of the observed stars.
+        :g2:        The g2 component of the shapes of the observed stars.
+        :T_model:   The size of the PSF model at the same locations as the stars.
+        :g1_model   The g1 component of the PSF model at these locations.
+        :g2_model   The g2 component of the PSF model at these locations.
+        :dT:        The size residual, T - T_model
+        :dg1:       The g1 residual, g1 - g1_model
+        :dg2:       The g2 residual, g2 - g2_model
+    """
+    def __init__(self, bins_size=10, bins_shape=10, file_name=None, logger=None):
         """
+        :param bins_size:   Number of bins for size histograms. [default: 10]
+        :param bins_shape:  Number of bins for shape histograms. [default: 10]
```

Comment:
It is worth pointing out (now that I think about this) that these can actually also be numpy arrays specifying bin edges as per plt.hist